book/src/passive-stake-delegation-and-rewards.md

```diff
@@ -0,0 +1,149 @@
+# Stake Delegation and Reward
+
+This design proposal focuses on the software architecture for the on-chain
+voting and staking programs.  Incentives for staking is covered in [staking
+rewardsd](staking-rewards.md).
+
+The current architecture requires an active vote for each delegated stake from
+the validator, and therefore does not scale to allow for replicator clients
+automatically delegate their rewards.
+ 
+The design proposes a new set of programs for voting and stake delegation, The
+proposed programs allow many stakes to passively earn rewards with a single
+validator vote without permission from the validator.
+
+## Current Design Problems
+
+In the current design each staker creates their own VoteState, and assigns a
+**delegate** in the VoteState that can submit votes.  Since the validator has to
+actively vote for each stake delegated to it, it can censor stakes by not voting
+for them.
+
+The number of votes is equal to the number of stakers, and not the number of
+validators.  Replicator clients are expected to automatically delegate their
+rewards, and therefore the number of stakes is expected to be large compared to
+the number of validators in a long running cluster.
+
+## Proposed changes to the current design.
+
+The general idea is that instead of the staker, the validator will own the
+VoteState program. In this proposal the VoteState program is there to track
+validator votes, count validator generated credits and to provide any
+additional validator specific state.  The VoteState program is not aware of any
+stakes delegated to it, and has no staking weight.
+
+The rewards generated are proportional to the amount of lamports staked.  In
+this proposal stake state is stored as part of the RewardsState program. This
+program is owned by the staker only.  Lamports stored in this program are the
+stake.  Unlike the current design, this program contains a new field to indicate
+which VoteState program the stake is delegated to.
+
+### New VoteState
+
+VoteState is the current state of all the votes the **delegate** has submitted
+to the bank.  VoteState contains the following state information:
+
+* votes - The submitted votes data structure.
+
+* credits - The total number of rewards this vote program generated over its
+lifetime.
+
+* root\_slot - The last slot to reach the full lockout commitment necessary for
+rewards.
+
+* commission - The commission taken by this VoteState for any rewards claimed by
+staker's RewardsState accounts.
+
+* lamports - The accumulated lamports from the commission.  These do not count as
+stakes.
+
+* `authorized_voter_id` - Only this identity is authorized to submit votes.
+
+### New RewardsState
+
+RewardsState is the current delegation preference of the **staker**. RewardsState
+contains the following state information:
+
+* lamports - The staked lamports.
+
+* `delegate` - The pubkey of the VoteState instance the lamports are
```

Comment:
This is confusing because there's currently a `delegate_id` that's used to delegate stake to a validator to determine the leader schedule. How about `voter_id`?